src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/SignInServiceTests.cs

```diff
@@ -26,685 +26,387 @@ public class SignInServiceTests
+            Assert.Equal(firstName, activatedUser.FirstName);
+            Assert.Equal(lastName, activatedUser.LastName);
+            Assert.True(activatedUser.Active);
+            Assert.Null(activatedUser.SoftDeleted);
+            Assert.Null(activatedUser.DeletedById);
+            Assert.Equal(invitedUser.RoleId, activatedUser.RoleId);
+            activatedUser.Created.AssertUtcNow();
+            Assert.Equal(activatedUser.CreatedById, activatedUser.CreatedById);
+        }
+    }
+
+    [Fact]
+    public async Task RegisterOrSignIn_HasAspNetUser_SuccessfulLogin()
+    {
+        var aspNetUser = new ApplicationUser
+        {
+            Id = Guid.NewGuid().ToString(),
+            Email = "test@test.com",
+            UserName = "test@test.com",
+            FirstName = "existing",
+            LastName = "user",
+        };
+
+        var usersAndRolesDbContextId = Guid.NewGuid().ToString();
+
+        await using (var usersAndRolesDbContext = InMemoryUserAndRolesDbContext(usersAndRolesDbContextId))
+        {
+            usersAndRolesDbContext.Users.Add(aspNetUser);
+            await usersAndRolesDbContext.SaveChangesAsync();
         }
 
-        [Fact]
-        public async Task RegisterOrSignIn_UserInviteMissingCreatedById_SetsUserCreatedByIdToDeletedUserPlaceholder()
+        var userService = new Mock<IUserService>(Strict);
+        userService
+            .Setup(mock => mock.GetProfileFromClaims())
+            .Returns(
+                new UserProfileFromClaims(
+                    Email: aspNetUser.Email,
+                    FirstName: aspNetUser.FirstName,
+                    LastName: aspNetUser.LastName
+                )
+            );
+
+        var userManager = MockUserManager();
+        userManager.Setup(mock => mock.FindByEmailAsync(aspNetUser.Email)).ReturnsAsync(aspNetUser);
+
+        await using (var usersAndRolesDbContext = InMemoryUserAndRolesDbContext(usersAndRolesDbContextId))
         {
-            var email = "test@test.com";
-            var firstName = "Bill";
-            var lastName = "Piper";
-
-            var userInvite = new UserInvite
-            {
-                Email = email,
-                Role = new IdentityRole
+            var service = SetupService(
+                usersAndRolesDbContext: usersAndRolesDbContext,
+                userService: userService.Object,
+                userManager: userManager.Object
+            );
+
+            var result = await service.RegisterOrSignIn();
+
+            var signInResponse = result.AssertRight();
+
+            Assert.Equal(LoginResult.LoginSuccess, signInResponse.LoginResult);
+            Assert.Equal(Guid.Parse(aspNetUser.Id), signInResponse.UserProfile!.Id);
+            Assert.Equal(aspNetUser.FirstName, signInResponse.UserProfile.FirstName);
+        }
+
+        VerifyAllMocks(userService, userManager);
+    }
+
+    [Fact]
+    public async Task RegisterOrSignIn_SoftDeletedUser_ReturnsNoInviteResult()
```

Comment:
I think you could add a test that checks an exception is thrown if `userManager.AddToRoleAsync` returns errors.

Well... you did ask for a review of the tests, to check for permutations that might be missing 🎃